Branch-SDK-Tests/BNCDeviceInfoTests.m

```diff
@@ -0,0 +1,145 @@
+//
+//  BNCDeviceInfoTests.m
+//  Branch-SDK-Tests
+//
+//  Created by Ernest Cho on 11/21/19.
+//  Copyright © 2019 Branch, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BNCDeviceInfo.h"
+
+@interface BNCDeviceInfoTests : XCTestCase
+@property (nonatomic, strong, readwrite) BNCDeviceInfo *deviceInfo;
+@end
+
+@implementation BNCDeviceInfoTests
+
+- (void)setUp {
+    self.deviceInfo = [BNCDeviceInfo new];
```

Comment:
```
[BNCDeviceInfo new];
```

Is this deviceInfo expensive to build or is it singleton? 
If there is no particular reason, can we just create them on each test case, given that some of test case involves in mutation of this object?

Reply:
It's relatively cheap and it's not a singleton. 

I think you're thinking of `+ (void)setUp`, which is one time for the whole test suite. ` - (void)setUp` runs before every test.